test/support/utils_sycl.h

```diff
@@ -36,6 +36,20 @@
 namespace TestUtils
 {
 
+#if __cplusplus >= 201703L
+#   if __SYCL_2020_NOINIT_MACRO
+#       define __noinit sycl::no_init
```

Comment:
I wouldn't suggest to use `__`. This is mostly for internal using. However, it's not a blocking comment